base/acme/src/org/dogtagpki/acme/validator/DNS01Validator.java

```diff
@@ -0,0 +1,158 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.acme.validator;
+
+import java.net.URI;
+import java.security.MessageDigest;
+import java.util.Hashtable;
+
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
+import javax.ws.rs.core.UriInfo;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang.RandomStringUtils;
+import org.dogtagpki.acme.ACMEAuthorization;
+import org.dogtagpki.acme.ACMEChallenge;
+import org.dogtagpki.acme.ACMEError;
+import org.dogtagpki.acme.ACMEIdentifier;
+
+/**
+ * @author Endi S. Dewata
+ */
+public class DNS01Validator extends ACMEValidator {
```

Comment:
Yes, it's just a matter of finding the right balance of what should be in base class and subclasses. So far we only have two validators, so I tend not to impose hard structure early on because it might impede future validators. But we can certainly refactor the code again anytime.